components/style/values/animated/effects.rs

```diff
@@ -0,0 +1,202 @@
+            GenericFilter::Grayscale(factor) => GenericFilter::Grayscale(factor),
+            GenericFilter::HueRotate(factor) => GenericFilter::HueRotate(factor),
+            GenericFilter::Invert(factor) => GenericFilter::Invert(factor),
+            GenericFilter::Opacity(factor) => GenericFilter::Opacity(factor),
+            GenericFilter::Saturate(factor) => GenericFilter::Saturate(factor),
+            GenericFilter::Sepia(factor) => GenericFilter::Sepia(factor),
+            GenericFilter::DropShadow(shadow) => {
+                GenericFilter::DropShadow(shadow.into())
+            },
+            #[cfg(feature = "gecko")]
+            GenericFilter::Url(url) => GenericFilter::Url(url),
+        }
+    }
+}
+
+impl From<Filter> for ComputedFilter {
+    #[inline]
+    fn from(filter: Filter) -> Self {
+        match filter {
+            GenericFilter::Blur(angle) => GenericFilter::Blur(angle),
+            GenericFilter::Brightness(factor) => GenericFilter::Brightness(factor),
+            GenericFilter::Contrast(factor) => GenericFilter::Contrast(factor),
+            GenericFilter::Grayscale(factor) => GenericFilter::Grayscale(factor),
+            GenericFilter::HueRotate(factor) => GenericFilter::HueRotate(factor),
+            GenericFilter::Invert(factor) => GenericFilter::Invert(factor),
+            GenericFilter::Opacity(factor) => GenericFilter::Opacity(factor),
+            GenericFilter::Saturate(factor) => GenericFilter::Saturate(factor),
+            GenericFilter::Sepia(factor) => GenericFilter::Sepia(factor),
+            GenericFilter::DropShadow(shadow) => {
+                GenericFilter::DropShadow(shadow.into())
+            },
+            #[cfg(feature = "gecko")]
+            GenericFilter::Url(url) => GenericFilter::Url(url.clone())
+        }
+    }
+}
+
+impl From<ComputedDropShadow> for DropShadow {
+    #[cfg(not(feature = "gecko"))]
+    #[inline]
+    fn from(shadow: ComputedDropShadow) -> Self {
+        match shadow {}
+    }
+
+    #[cfg(feature = "gecko")]
+    #[inline]
+    fn from(shadow: ComputedDropShadow) -> Self {
+        DropShadow {
+            color: shadow.color.into(),
+            horizontal: shadow.horizontal,
+            vertical: shadow.vertical,
+            blur: shadow.blur,
+        }
+    }
+}
+
+impl From<DropShadow> for ComputedDropShadow {
+    #[cfg(not(feature = "gecko"))]
+    #[inline]
+    fn from(shadow: DropShadow) -> Self {
+        match shadow {}
+    }
+
+    #[cfg(feature = "gecko")]
+    #[inline]
+    fn from(shadow: DropShadow) -> Self {
+        ComputedDropShadow {
+            color: shadow.color.into(),
+            horizontal: shadow.horizontal,
+            vertical: shadow.vertical,
+            blur: shadow.blur,
+        }
+    }
+}
+
+impl Animatable for DropShadow {
+    #[cfg(not(feature = "gecko"))]
+    #[inline]
+    fn add_weighted(&self, _other: &Self, _self_portion: f64, _other_portion: f64) -> Result<Self, ()> {
+        match *self {}
```

Comment:
This is quite nice actually, didn't expect it to compile without throwing `unreachable` at it.